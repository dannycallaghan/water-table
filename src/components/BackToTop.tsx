function BackToTop() {
  const handleClick = () => {
    window.scrollTo({ top: 0, behavior: 'smooth' });
  };

  return (
    <div
      className="flex cursor-pointer flex-row items-center gap-1"
      onClick={handleClick}
    >
      <p className="text-[0.7rem] uppercase text-gray-400">Back to top</p>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 24 24"
        strokeWidth={1.5}
        stroke="currentColor"
        className="size-3 text-gray-400"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          d="M4.5 10.5 12 3m0 0 7.5 7.5M12 3v18"
        />
      </svg>
    </div>
  );
}

export default BackToTop;
