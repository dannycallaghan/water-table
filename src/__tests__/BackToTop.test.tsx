import { expect, test, describe } from 'vitest';
import { render } from '@testing-library/react';
import BackToTop from '../components/BackToTop.tsx';

describe('BackToTop', () => {
  test('renders correctly', () => {
    const { asFragment } = render(<BackToTop />);
    expect(asFragment()).toMatchSnapshot();
  });
});
