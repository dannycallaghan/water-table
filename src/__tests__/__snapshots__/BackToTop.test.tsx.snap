// Vitest Snapshot v1

exports[`BackToTop > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex cursor-pointer flex-row items-center gap-1"
  >
    <p
      class="text-[0.7rem] uppercase text-gray-400"
    >
      Back to top
    </p>
    <svg
      class="size-3 text-gray-400"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.5 10.5 12 3m0 0 7.5 7.5M12 3v18"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </div>
</DocumentFragment>
`;
